src/proxy.rs

```diff
@@ -373,6 +373,16 @@ where
     }
 }
 
+impl<T, P> Debug for ConstrainedFloat<T, P>
+where
+    T: Float + Primitive + Debug,
+    P: FloatConstraint<T>,
+{
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
```

Comment:
Is it possible to elide the lifetime parameter of `Formatter` here (as seen in the `Display` implementation)?

```suggestion
    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
```

Reply:
Yes, but lifetime elision in paths is now deprecated in Rust 2018. See:

https://doc.rust-lang.org/edition-guide/rust-2018/ownership-and-lifetimes/the-anonymous-lifetime.html

> Therefore, starting with Rust 2018, it is deprecated to leave off the lifetime parameters for non-reference-types (types other than & and &mut).

There is also a compiler lint for that, however the rustc team decided to leave it at `allow` per default -- it can be turned into a warning via:

```rust
#![warn(elided_lifetimes_in_paths)]
```

Do you want me to change it anyway or perhaps un-elide the other usages to keep things consistent?